Testing/SystemTests/tests/analysis/PearlPowderDiffractionScriptTest.py

```diff
@@ -0,0 +1,190 @@
+                          "PEARL/DataOut/PRL92476_92479-0.gss"))
+        self._clean_up_files(filenames, DIRS)
+
+
+# ======================================================================
+# work horse
+class LoadTests(unittest.TestCase):
+    wsname = "__LoadTest"
+    cleanup_names = []
+
+    def tearDown(self):
+        self.cleanup_names.append(self.wsname)
+        for name in self.cleanup_names:
+            try:
+                AnalysisDataService.remove(name)
+            except KeyError:
+                pass
+        self.cleanup_names = []
+
+        # ============================ Success =============================
+
+    def test_vanadium_workspace(self):
+        for i in range(1, 15):
+            van_file = mtd["van" + str(i)]
+            self.assertTrue(isinstance(van_file, MatrixWorkspace))
+            self.assertEquals(1, van_file.getNumberHistograms())
+            self.assertEquals(4310, van_file.blocksize())
+
+        # check random values, places variables doesnt seem to work when
+        van1 = mtd["van1"]
+        self.assertAlmostEqual(1690.17031579, van1.readX(0)[199], places=DIFF_PLACES)
+        van4 = mtd["van4"]
+        self.assertAlmostEqual(6.5605707095e-06, van4.readY(0)[4000], places=DIFF_PLACES)
+        van7 = mtd["van7"]
+        self.assertAlmostEquals(1.03406081396e-08, van7.readE(0)[50], places=DIFF_PLACES)
+        van10 = mtd["van10"]
+        self.assertAlmostEquals(11667.996525642744, van10.readX(0)[3420], places=DIFF_PLACES)
+        van14 = mtd["van14"]
+        self.assertAlmostEqual(3.39159292684e-05, van14.readY(0)[187], places=DIFF_PLACES)
+
+    def test_rdata_workspace(self):
+        for i in range(1, 15, 3):
+            rfile = mtd["rdata" + str(i)]
+            self.assertTrue(isinstance(rfile, MatrixWorkspace))
+            self.assertEquals(1, rfile.getNumberHistograms())
+            self.assertEquals(4310, rfile.blocksize())
+            self.assertIn("glucose+Pb 6 tns in ZTA anvils", rfile.getTitle())
+
+            van_file = mtd["van" + str(i)]
+            self.assertAlmostEquals(van_file.readX(0)[i], rfile.readX(0)[i], places=DIFF_PLACES)
+
+        focus_file = mtd["focus"]
+        self.assertTrue(isinstance(focus_file, MatrixWorkspace))
+        self.assertEquals(14, focus_file.getNumberHistograms())
+        self.assertEquals(4310, focus_file.blocksize())
+        self.assertIn("glucose+Pb 6 tns in ZTA anvils", rfile.getTitle())
+
+    def test_mod_files(self):
+        mod_group_table = mtd["PRL92476_92479"]
+        self.assertTrue("PRL92476_92479_mods1-9_1" in mod_group_table[0].getName())
+
+        rand_range = random.randint(1, 4)
+        for i in range(rand_range, 10, 2):
+            self.assertTrue("PRL92476_92479_mod" + str(i) in mod_group_table[i].getName())
+            self.assertEquals(4310, mod_group_table[i].blocksize())
+            ind_mod_file = mtd["PRL92476_92479_mod" + str(i)]
+            rand_test = random.randint(1, 4300)
```

Comment:
And here you use `randint` to pick one spectrum at random? This is also non-deterministic testing. Now it fails, now it passes. It can lead to nightmares.
If you want to pick one or a few random indices (but always the same in all test repetitions) then you could use `seed()`. I'd pick several spectra in any case, and not random if there's no particular reason to use `random`.